### ISSUE TITLE
Portfolio Beta and ADS 2019

### ISSUE THREAD

**Issue by jespersh** [microsoft/featuretimeline#127]
**Describe the bug**
This is a question about if the Portfolio Beta works with Azure DevOps Server. I am unable to get projects listed and fails on some analytics request

**Browser**
Choose a browser
[ ] IE11
[X] Chrome
[X] FireFox
[ ] Safari

**Version**
[X] Azure DevOps Server 2019

**Screenshots**
![image](https://user-images.githubusercontent.com/1004198/61364908-ffab2100-a886-11e9-9b87-7b6dfeacbdef.png)

![image](https://user-images.githubusercontent.com/1004198/61364965-1b162c00-a887-11e9-8b3c-1d3a0e276f97.png)

**Comment by navneet-g** [comment]
@ezambra please take a look.

**Comment by ezambra** [comment]
Thanks for reporting @jespersh ... I'll start investigating, and let you know when I have more info.

**Comment by Sebbl22** [comment]
Exact same problem here. Identical error message, identical HTTP 404 code in browser dev tools.
On-premise Azure DevOps Server 2019 (17.143.28912.1).

**Comment by Christ0phH** [comment]
+1 on the problem. I have to say the Issue name is not very good, i was about to open a new issue regarding the errors...

Choose a browser
[X] IE11 **no important**
[X] Chrome  **75.0.3770.90**
[X] FireFox **60.8.0esr**
[ ] Safari **(not tested)**


Version
 TFS 2018 Version 16.131.28601.4

``
{"exceptionMessage":"{\n \"readyState\": 0,\n \"status\": 0,\n \"statusText\": \"error\"\n}"}
``


Console Output on my end:
```
PortfolioPlanning.html:1 Access to XMLHttpRequest at 'https://analytics.dev.azure.com/Team%20Foundation%20Server/_odata/v3.0-preview/Projects?$select=ProjectSK,ProjectName' from origin 'null' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.
ms-devlabs.workitem-feature-timeline-extension.workitem-portfolio-planning:1 Unchecked runtime.lastError: The message port closed before a response was received.
```

**Comment by ezambra** [comment]
After investigation, the problem is caused by a missing feature in OData service for DevOps Server 2019.
 
Portfolio Plans uses the `Descendants` clause included in OData version `3.0-preview`, but unfortunately `Descendants` is not enabled yet. This should be fully enabled in a future DevOps Server release. In the meanwhile, I've pushed an update to the extension to disable `Portfolio Plans` hub is DevOps Server for now. I'll also update the extension's documentation to reflect that `Portfolio Plans` is still not supported in DevOps Server.

**Comment by jespersh** [comment]
Thank you. I didn't expect it was ready for ADS either. Looking forward to what it will be like to work with
